ignite/utils.py

```diff
@@ -41,11 +41,13 @@ def apply_to_type(
     if isinstance(input_, (str, bytes)):
         return input_
     if isinstance(input_, collections.Mapping):
-        return type(input_)({k: apply_to_type(sample, input_type, func) for k, sample in input_.items()})
+        return cast(Callable, type(input_))(
+            {k: apply_to_type(sample, input_type, func) for k, sample in input_.items()}
+        )
     if isinstance(input_, tuple) and hasattr(input_, "_fields"):  # namedtuple
-        return type(input_)(*(apply_to_type(sample, input_type, func) for sample in input_))
+        return cast(Callable, type(input_))(*(apply_to_type(sample, input_type, func) for sample in input_))
```

Comment:
I was wondering if this is the only way to do this and have mypy happy ?

Reply:
at least I couldn't find any better solution. Somehow `mypy` doesn't understand what is happening there. I have to admit, that I also didn't understand at the beginning, what that weird syntax is doing and then I figured it out 😄

Reply:
Yeah, I agree that this looks weird but it is a way to recreate initial structure and apply func on its entries...